aws-sdk-core/lib/aws-sdk-core/plugins/s3_redirects.rb

```diff
@@ -13,7 +13,7 @@ class Handler < Seahorse::Client::Handler
 
         def call(context)
           response = @handler.call(context)
-          if context.http_response.status_code == 307
+          if context.http_response.status_code == 307 or 302
```

Comment:
This if statement will *always* return true. You would instead write something similar to:

```ruby
status_code = context.http_response.status_code
if status_code == 307 || status_code == 303
  # do things
end
```